code/HabitTracker/app/src/main/java/com/example/habittracker/activities/eventlist/LocationActivity.java

```diff
@@ -63,20 +67,28 @@ protected void onCreate(Bundle savedInstanceState) {
             textView.setText("Please choose location");
         }
 
-
         textView.setText("Latitude: " + latitude + "\n" +
                 "Longitude: " + longitude + "\n" +
                 "Address: " + address);
 
     }
 
     /**
-     * Override onClick method for passing intent
+     * switches to the Maps Activity
      * @param v
      */
     public void onClick(View v) {
         Intent intent = new Intent(this, MapsActivity.class);
         startActivity(intent);
         Toast.makeText(LocationActivity.this, "Long click to choose location", Toast.LENGTH_LONG).show();
     }
+
+    /**
+     * Switches back from Location Activity
+     */
+    public void onBackPressed() {
+        super.onBackPressed();
+        Intent myIntent = new Intent(LocationActivity.this, HomeActivity.class);
+        startActivity(myIntent);
+    }
```

Comment:
Just wondering how this is integrated with the rest of the project. Shouldn't the back button go back to the previous screen ie. the habit event fragment, not starting the home activity?

Reply:
that is a temporary fix going back to the fragment was somehow causing crashes-- this is something John and I will fix during reading week